cpp/src/ktruss/ktruss.cu

```diff
@@ -0,0 +1,115 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * ---------------------------------------------------------------------------*
+ * @brief KTruss implementation
+ *
+ * @file ktruss.cu
+ * --------------------------------------------------------------------------*/
+
+
+#include <cugraph.h>
+#include "utilities/error_utils.h"
+#include <Hornet.hpp>
+#include "Static/KTruss/KTruss.cuh"
+#include <StandardAPI.hpp>
+#include <rmm_utils.h>
+#include <nvgraph_gdf.h>
+
+using namespace hornets_nest;
+
+gdf_error ktruss_max_impl(gdf_graph *graph,
+                          int *k_max) {
+  gdf_error err = gdf_add_adj_list(graph);
+  if (err != GDF_SUCCESS)
+    return err;
+  int * src = static_cast<int*>(graph->edgeList->src_indices->data);
+  int * dst = static_cast<int*>(graph->edgeList->dest_indices->data);
+  using HornetGraph = hornet::gpu::Hornet<int>;
+  using UpdatePtr   = ::hornet::BatchUpdatePtr<int, hornet::EMPTY, hornet::DeviceType::DEVICE>;
+  using Update      = ::hornet::gpu::BatchUpdate<int>;
+
+  UpdatePtr ptr(graph->edgeList->src_indices->size, src, dst);
+  Update batch(ptr);
+  HornetGraph hnt(graph->numberOfVertices);
+  hnt.insert(batch);
+  //Use hornet
+
+/*
+  int *h_tempOffset,*h_tempIndices;
+  h_tempOffset = (int*)malloc((graph->numberOfVertices+1)*sizeof(int));
+  h_tempIndices = (int*)malloc((graph->adjList->indices->size)*sizeof(int));
+
+  cudaMemcpy(h_tempOffset,static_cast<int*>(graph->adjList->offsets->data),(graph->numberOfVertices+1)*sizeof(int),cudaMemcpyDeviceToHost);
+  cudaMemcpy(h_tempIndices,static_cast<int*>(graph->adjList->indices->data),(graph->adjList->indices->size)*sizeof(int),cudaMemcpyDeviceToHost);
+
+  // HornetInit init(graph->numberOfVertices, graph->adjList->indices->size,
+  //     static_cast<int*>(graph->adjList->offsets->data),
+  //     static_cast<int*>(graph->adjList->indices->data));
+  using HornetInit  = hornet::HornetInit<int>;
+
+
+  HornetInit init(graph->numberOfVertices, graph->adjList->indices->size,
+      h_tempOffset,
+      h_tempIndices);
+
+  HornetGraph hnt(init);
+
+  free(h_tempOffset);
+  free(h_tempIndices);
+*/
+
+  
+  KTruss kt(hnt);
+
+  kt.init();
+  kt.reset();
+
+  kt.createOffSetArray();
+  kt.setInitParameters(4, 8, 2, 64000, 32);
+  kt.reset(); 
+  kt.sortHornet();
+
+  kt.run();
+
+  // *k_max = 12;
+
+  *k_max = kt.getMaxK();
+
+  kt.release();
+  
+  return GDF_SUCCESS;
+}
+
+gdf_error gdf_k_truss_max(gdf_graph *graph,
```

Comment:
FYI we are getting rid of `gdf_error` return type, `gdf_` prefix and adding `cugraph::` namespace on top of API functions.
Here is how `k_core` looks like after the changes for instance: https://github.com/afender/cugraph/blob/c85e2fbd55491d037e476f3dd529da3cd3654240/cpp/src/cores/core_number.cu#L204

Reply:
@afender, I have made those modifications now in the C/CUDA code. Does the python code need to catch these errors? Or is that user responsibility?

Reply:
> Does the python code need to catch these errors?

I see you already added `except +` in the bindings so python will get the C++ exception and make it a python one :+1: .

Reply:
I updated today based on these comments.